server/src/main/java/org/elasticsearch/rest/action/document/RestIndexAction.java

```diff
@@ -104,11 +106,12 @@ public RestChannelConsumer prepareRequest(RestRequest request, final NodeClient
 
     @Override
     public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
+        ReleasableBytesReference source = request.tryRetainedContent();
```

Comment:
I'd rather we used something like `org.elasticsearch.core.RefCounted#mustIncRef` here to document that we expect `request` not to be closed at the point this method is called. Indeed I'd rather we used `mustIncRef` directly and avoided non-refcount-neutral methods like `tryRetainedContent()`.